### ISSUE TITLE
Bootstrap service off after update

### ISSUE THREAD

**Issue by kanecko** [rockstor/rockstor-core#2488]
Discussion : 
 https://forum.rockstor.com/t/bootstrap-service-disabled-after-v4-5-4-0-upgrade/8626/7

The issue is that the rockstor-bootstrap service doesn't starts after the update finishes successfully.

Hello all.

I can reproduce this bug every update. It doesnt matter if in 15.3 or 15.4 version after a fresh install.
This time I have saved the logs before turning it on again.

UI:
![image](https://user-images.githubusercontent.com/5829467/215167518-019348dc-3fe0-4ff1-99de-3ad7c37273f0.png)

systemctl:
```
● rockstor.service - Rockstor startup script
     Loaded: loaded (/usr/lib/systemd/system/rockstor.service; disabled; vendor preset: disabled)
     Active: active (running) since Thu 2023-01-26 16:38:45 CET; 19h ago
   Main PID: 10742 (supervisord)
      Tasks: 11
     CGroup: /system.slice/rockstor.service
             ├─ 10742 /opt/rockstor/.venv/bin/python /opt/rockstor/.venv/bin/supervisord -c /opt/rockstor/etc/supervisord.conf
             ├─ 10758 /opt/rockstor/.venv/bin/python /opt/rockstor/.venv/bin/gunicorn --bind=127.0.0.1:8000 --pid=/run/gunicorn.pid --workers=2 --log-file=/opt/rockstor/var/log/gunicorn.log --pythonpath=/opt/rockstor/src/rockstor --timeout=120 --graceful-timeout=120 wsgi:application
             ├─ 10759 /opt/rockstor/.venv/bin/python /opt/rockstor/.venv/bin/data-collector
             ├─ 10760 /opt/rockstor/.venv/bin/python /opt/rockstor/.venv/bin/django-admin run_huey --workers 2 --worker-type thread --logfile /opt/rockstor/var/log/huey.log
             ├─ 10774 /opt/rockstor/.venv/bin/python /opt/rockstor/.venv/bin/gunicorn --bind=127.0.0.1:8000 --pid=/run/gunicorn.pid --workers=2 --log-file=/opt/rockstor/var/log/gunicorn.log --pythonpath=/opt/rockstor/src/rockstor --timeout=120 --graceful-timeout=120 wsgi:application
             └─ 10778 /opt/rockstor/.venv/bin/python /opt/rockstor/.venv/bin/gunicorn --bind=127.0.0.1:8000 --pid=/run/gunicorn.pid --workers=2 --log-file=/opt/rockstor/var/log/gunicorn.log --pythonpath=/opt/rockstor/src/rockstor --timeout=120 --graceful-timeout=120 wsgi:application

Jan 26 16:38:45 rockstor systemd[1]: Started Rockstor startup script.
Jan 26 16:38:45 rockstor poetry[10742]: 2023-01-26 16:38:45,882 INFO Set uid to user 0 succeeded
Jan 26 16:38:45 rockstor poetry[10742]: 2023-01-26 16:38:45,896 INFO RPC interface 'supervisor' initialized
Jan 26 16:38:45 rockstor poetry[10742]: 2023-01-26 16:38:45,896 INFO supervisord started with pid 10742
Jan 26 16:38:46 rockstor poetry[10742]: 2023-01-26 16:38:46,899 INFO spawned: 'gunicorn' with pid 10758
Jan 26 16:38:46 rockstor poetry[10742]: 2023-01-26 16:38:46,901 INFO spawned: 'data-collector' with pid 10759
Jan 26 16:38:46 rockstor poetry[10742]: 2023-01-26 16:38:46,903 INFO spawned: 'ztask-daemon' with pid 10760
Jan 26 16:38:48 rockstor poetry[10742]: 2023-01-26 16:38:48,907 INFO success: data-collector entered RUNNING state, process has stayed up for > than 2 seconds (startsecs)
Jan 26 16:38:51 rockstor poetry[10742]: 2023-01-26 16:38:51,912 INFO success: gunicorn entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
Jan 26 16:38:56 rockstor poetry[10742]: 2023-01-26 16:38:56,919 INFO success: ztask-daemon entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)

○ rockstor-bootstrap.service - Rockstor bootstrapping tasks
     Loaded: loaded (/usr/lib/systemd/system/rockstor-bootstrap.service; enabled; vendor preset: enabled)
     Active: inactive (dead) since Thu 2023-01-26 16:38:26 CET; 19h ago
   Main PID: 27010 (code=exited, status=0/SUCCESS)

Jan 26 14:23:24 rockstor systemd[1]: Starting Rockstor bootstrapping tasks...
Jan 26 14:23:30 rockstor bootstrap[27010]: BTRFS device scan complete
Jan 26 14:23:30 rockstor bootstrap[27010]: Bootstrapping complete
Jan 26 14:23:30 rockstor bootstrap[27010]: Running qgroup cleanup. /opt/rockstor/.venv/bin/qgroup-clean
Jan 26 14:23:30 rockstor bootstrap[27010]: Running qgroup limit maxout. /opt/rockstor/.venv/bin/qgroup-maxout-limit
Jan 26 14:23:30 rockstor systemd[1]: Finished Rockstor bootstrapping tasks.
Jan 26 16:38:26 rockstor systemd[1]: rockstor-bootstrap.service: Deactivated successfully.
Jan 26 16:38:26 rockstor systemd[1]: Stopped Rockstor bootstrapping tasks.
```

journalctl:
```
Jan 26 14:18:57 rockstor systemd[1]: rockstor-bootstrap.service: Deactivated successfully.
Jan 26 14:18:57 rockstor systemd[1]: Stopped Rockstor bootstrapping tasks.
Jan 26 14:19:11 rockstor systemd[1]: Starting Rockstor bootstrapping tasks...
Jan 26 14:19:17 rockstor bootstrap[21358]: BTRFS device scan complete
Jan 26 14:19:17 rockstor bootstrap[21358]: Bootstrapping complete
Jan 26 14:19:17 rockstor bootstrap[21358]: Running qgroup cleanup. /opt/rockstor/.venv/bin/qgroup-clean
Jan 26 14:19:17 rockstor bootstrap[21358]: Running qgroup limit maxout. /opt/rockstor/.venv/bin/qgroup-maxout-limit
Jan 26 14:19:17 rockstor systemd[1]: Finished Rockstor bootstrapping tasks.
Jan 26 14:20:42 rockstor systemd[1]: rockstor-bootstrap.service: Deactivated successfully.
Jan 26 14:20:42 rockstor systemd[1]: Stopped Rockstor bootstrapping tasks.
Jan 26 14:20:52 rockstor systemd[1]: Starting Rockstor bootstrapping tasks...
Jan 26 14:20:59 rockstor bootstrap[24265]: BTRFS device scan complete
Jan 26 14:20:59 rockstor bootstrap[24265]: Bootstrapping complete
Jan 26 14:20:59 rockstor bootstrap[24265]: Running qgroup cleanup. /opt/rockstor/.venv/bin/qgroup-clean
Jan 26 14:20:59 rockstor bootstrap[24265]: Running qgroup limit maxout. /opt/rockstor/.venv/bin/qgroup-maxout-limit
Jan 26 14:20:59 rockstor systemd[1]: Finished Rockstor bootstrapping tasks.
Jan 26 14:23:11 rockstor systemd[1]: rockstor-bootstrap.service: Deactivated successfully.
Jan 26 14:23:11 rockstor systemd[1]: Stopped Rockstor bootstrapping tasks.
Jan 26 14:23:24 rockstor systemd[1]: Starting Rockstor bootstrapping tasks...
Jan 26 14:23:30 rockstor bootstrap[27010]: BTRFS device scan complete
Jan 26 14:23:30 rockstor bootstrap[27010]: Bootstrapping complete
Jan 26 14:23:30 rockstor bootstrap[27010]: Running qgroup cleanup. /opt/rockstor/.venv/bin/qgroup-clean
Jan 26 14:23:30 rockstor bootstrap[27010]: Running qgroup limit maxout. /opt/rockstor/.venv/bin/qgroup-maxout-limit
Jan 26 14:23:30 rockstor systemd[1]: Finished Rockstor bootstrapping tasks.
Jan 26 16:38:26 rockstor systemd[1]: rockstor-bootstrap.service: Deactivated successfully.
Jan 26 16:38:26 rockstor systemd[1]: Stopped Rockstor bootstrapping tasks.
```

Going from zypper log, I started update at 2023-01-26 16:38:29

```
2023-01-26 16:38:29|command|root@rockstor|'/usr/bin/zypper' '--non-interactive' 'install' 'rockstor'|
2023-01-26 16:38:31|install|rockstor|4.5.6-0|x86_64|root@rockstor|Rockstor-Testing|429fba2721dce92f9d27b9eb4977ef24042e9ba258186721ff2e9892667bf58a|
# 2023-01-26 16:38:34 Output of rockstor-4.5.6-0.x86_64.rpm %posttrans script:
# Retrieving Poetry metadata
```

**Comment by phillxnet** [comment]
@kanecko Thanks for creating this issue, much appreciated.

As discussed in the investigations prompted by your thread there we have the following anomaly in our rockstor update mechanism:

https://github.com/rockstor/rockstor-core/blob/5ea4b206d068cf095277b0fee6238d71d16dadb3/src/rockstor/system/pkg_mgmt.py#L466-L474

Where after an update we re-start the new (updated) instance via the temporary (self deleting) AT script our former instance uses to orchestrate our update. But we use the now dated method of starting rockstor.service. We have for some time now used instead rockstor-bootstrap.service. And we have more recently modified our OBS systemd-presets-branding-rockstor app to do just this:

https://build.opensuse.org/package/show/home:rockstor:branches:Base:System/systemd-presets-branding-rockstor

> Philip Guyton (rockstor) committed about 1 month ago (revision 16)
Enable our top service, which will, in-turn, start all others.

So it is very likely that we just need to re-start ourselves (in the AT script) in line with how we do it via normal boot-up (rockstor-bootstrap) and our prior initrocks setups etc did establish this as a systemd service that was enabled by default.

Repeating here, from my forum comments, to collect the most relevant info under this issue.

This should be an easy fix of simply writing in the rockstor-bootstrap service instead of the rockstor one, when we dymanicaly create our update AT script.

**Comment by phillxnet** [comment]
It is to be proposed in a pull request under development/testing that we modify our existing dynamically created AT script to read as follows:
```
sleep 10
/usr/bin/systemctl stop rockstor*
/usr/bin/find /opt/rockstor/ -name "*.pyc" -not -path "*/eggs/*" -type f -delete
/usr/bin/zypper --non-interactive refresh
/usr/bin/zypper --non-interactive install rockstor
/usr/bin/systemctl start rockstor-bootstrap
/bin/rm -f /tmp/tmpiSUGxj
```
Where the temp file name used is managed by our use of mkstemp() (unchanged for this issue).

There by substituting our prior start of the rockstor service with that of a rockstor-bootstrap start.

A test was performed using a modified (according to in-development/testing pr for this issue) 4.5.6-0 install resulting from our current installer. Where a local lan repository offered an zypper available rockstor package upgrade. A Web-UI upgrade was performed and completed as normal. However the services page there-after showed:
![bootstrap-servcie-on-after-upgrade](https://user-images.githubusercontent.com/2521585/218276229-df84ba23-0e2c-48ed-a0b9-6909ad036bcf.png)
i.e. a Web-UI indication of the rockstor-bootstrap service being 'on'.

We also have the following cli status:
```
buildvm:~ # systemctl status rockstor-bootstrap
● rockstor-bootstrap.service - Rockstor bootstrapping tasks
     Loaded: loaded (/usr/lib/systemd/system/rockstor-bootstrap.service; enabled; vendor preset: enabled)
     Active: active (exited) since Sat 2023-02-11 18:40:32 WET; 19min ago
   Main PID: 13420 (code=exited, status=0/SUCCESS)

Feb 11 18:40:24 buildvm systemd[1]: Starting Rockstor bootstrapping tasks...
Feb 11 18:40:32 buildvm bootstrap[13420]: BTRFS device scan complete
Feb 11 18:40:32 buildvm bootstrap[13420]: Bootstrapping complete
Feb 11 18:40:32 buildvm bootstrap[13420]: Running qgroup cleanup. /opt/rockstor/.venv/bin/qgroup-clean
Feb 11 18:40:32 buildvm bootstrap[13420]: Running qgroup limit maxout. /opt/rockstor/.venv/bin/qgroup-maxout-limit
Feb 11 18:40:32 buildvm systemd[1]: Finished Rockstor bootstrapping tasks.
buildvm:~ # date
Sat 11 Feb 19:00:18 WET 2023
```

Two installed Rock-ons (SmokePing & Web Service Discovery For Windows Networks) and an SMB share config were persistent over this in-place Web-UI initiated rockstor rpm upgrade (see below note for Samba service anomaly). And the Rockstor Web-UI header correctly indicated the new rpm version.

## Caveat
Of note however was that the SAMBA service was in an off state post update! This is still under investigation.

**Comment by phillxnet** [comment]
The above test was repeated without the Caveat observed this time: likely tester error on this one: edited prior comment accordingly.

**Comment by phillxnet** [comment]
@kanecko Thanks again for highlighting this issue, much appreciated.
Closing as:
Fixed by #2495
